File: mvn-advent-of-code/src/test/java/advent/twenty_seventeen/Day20ParticleSwarmTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day20ParticleSwarmTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day20ParticleSwarm("/twenty_seventeen/Day20-ParticleSwarm-input.txt");

        assertEquals(344, dp.getPart1Answer().intValue());
        assertEquals(404, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day20ParticleSwarm("/twenty_seventeen/Day20-ParticleSwarm-example#1.txt");

        assertEquals(0, dp.getPart1Answer().intValue());
        assertEquals(2, dp.getPart2Answer().intValue());
    }


    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day20ParticleSwarm("/twenty_seventeen/Day20-ParticleSwarm-example#2.txt");

        assertEquals(3, dp.getPart1Answer().intValue());
        assertEquals(1, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day20ParticleSwarmTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day20ParticleSwarmTest  extends FlatSpec with Matchers {
  "2017 Day 20 - Input" should "calculate answers" in {
    val dp = new Day20ParticleSwarm("/twenty_seventeen/Day20-ParticleSwarm-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day20ParticleSwarm("/twenty_seventeen/Day20-ParticleSwarm-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
